src/biogeophys/SoilHydrologyMod.F90

```diff
@@ -844,11 +845,15 @@ subroutine WaterTable(bounds, num_hydrologyc, filter_hydrologyc, num_urbanc, fil
              ! make consistent with how evap_grnd removed in infiltration
              h2osoi_liq(c,1) = h2osoi_liq(c,1) + (1._r8 - frac_h2osfc(c))*qflx_liqdew_to_top_layer(c) * dtime
              h2osoi_ice(c,1) = h2osoi_ice(c,1) + (1._r8 - frac_h2osfc(c))*qflx_soliddew_to_top_layer(c) * dtime
-             if (qflx_solidevap_from_top_layer(c)*dtime > h2osoi_ice(c,1)) then
+             if ((1._r8 - frac_h2osfc(c))*qflx_solidevap_from_top_layer(c)*dtime > h2osoi_ice(c,1)) then
                 qflx_solidevap_from_top_layer_save = qflx_solidevap_from_top_layer(c)
                 qflx_solidevap_from_top_layer(c) = h2osoi_ice(c,1)/dtime
```

Comment:
I think what should happen here is that the if-block should just contain an endrun and the other code should be removed because 1) the new limitation should correct his issue unless there is some kind of bug and 2) the flux/state updates in the if-block don't conserve energy and mass, even with the (1-frac_h2osfc).